doc/api-design.yml

```diff
@@ -0,0 +1,903 @@
+              schema:
+                type: array
+                items:
+                  type: string
+        '400':
+          description: Invalid input
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+  /question_answers:
+    get:
+      summary: Get question answers
+      tags:
+        - Questions
+      security:
+        - bearerAuth: []
+      parameters:
+        - name: word
+          in: query
+          required: true
+          schema:
+            type: string
+        - name: type
+          in: query
+          required: true
+          schema:
+            type: string
+            enum: [english_to_turkish, turkish_to_english, english_to_sense]
+      responses:
+        '200':
+          description: Question answers retrieved successfully
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  type: object
+                  properties:
+                    correct_answer:
+                      type: string
+                    wrong_answer1:
+                      type: string
+                    wrong_answer2:
+                      type: string
+                    wrong_answer3:
+                      type: string
+        '400':
+          description: Invalid input
+  /quizzes:
+    post:
+      summary: Create a new quiz
+      tags:
+        - Quizzes
+      security:
+        - bearerAuth: []
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                title:
+                  type: string
+                description:
+                  type: string
+                image:
+                  type: string
+                  format: base64
+                questions:
+                  type: array
+                  items:
+                    type: object
+                    properties:
+                      question_type:
+                        type: string
+                        enum: [english_to_turkish, turkish_to_english, english_to_sense]
+                      word:
+                        type: string
+                      correct_answer:
+                        type: string
+                      wrong_answers:
+                        type: array
+                        items:
+                          type: string
+              required:
+                - title
+                - questions
+      responses:
+        '201':
+          description: Quiz created successfully
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Quiz'
+        '400':
+          description: Invalid input
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+    get:
```

Comment:
It would be really beneficial to add a GET request for quizzes for specific difficulty levels, since we need to display the Quizzes only for one specific difficulty level in both sections of the Home Screen.

Reply:
it is possible with request parameter "difficulty"

Reply:
How can we obtain say quizzes for A1 level only? This corresponds to ELOs 0-500. Can we provide such a range as a parameter?

Reply:
GET /quizzes?difficulty=a1 works for that purpose. However it could be beneficial to retrieve quizzes by specifiying upper and lower elo limits especially to use for users who are at the borders of english levels. After implemention of basic retrievals, it will be easy to implement neccessary filtering operations.

Reply:
Yeap that was what I was confused about, thanks for letting me know!